### ISSUE TITLE
Swift 5.2 support (Xcode 11.4)

### ISSUE THREAD

**Issue by rjchatfield** [exyte/Macaw#668]
Macaw doesn't compile with Swift 5.2 included in Xcode 11.4 beta 1.

There are 4 compilation failures that I saw. This was in `Gradient.swift`:

<img width="731" alt="image" src="https://user-images.githubusercontent.com/5361118/73892294-a5c58380-48ca-11ea-8816-1b5beed3aa98.png">

Error message:

```
Overridden method 'equals' has generic signature <T where T : Gradient> which is incompatible with base method's generic signature <T where T : Fill>; expected generic signature to be <T where T : Fill>
```

Perhaps as a workaround to this to change the `T: Gradient` up to the expected `T: Fill` and the guard on the first line to cast it to the expected type, else `false`:

```swift
// Before:
override func equals<T>(other: T) -> Bool where T: Gradient {
    ...
}
```

```swift
// Suggestion:
override func equals<T>(other: T) -> Bool where T: Fill {
    guard let other = other as? Gradient else { 
        return false 
    }
    ...
}
```

**Comment by amarunko** [comment]
Hi, @rjchatfield, first of all, thank you for the suggestion, you are welcome to send a PR

**Comment by amarunko** [comment]
Yea, you are right this is an appropriate solution. I created a draft [PR](https://github.com/exyte/Macaw/pull/669) for new beta

**Comment by khlopko** [comment]
Hello! Any updates on that?

**Comment by premedios** [comment]
The proposed changes worked. Waiting on release of new version.

**Comment by ystrot** [comment]
Merged into master.
